cherrypy/test/test_bus.py

```diff
@@ -261,10 +261,15 @@ def g():  # pylint: disable=invalid-name
 
     # The last message will mention an indeterminable thread name; ignore
     # it
-    assert (log_tracker.log_entries[:-1] ==
-            ['Bus STOPPING', 'Bus STOPPED',
-             'Bus EXITING', 'Bus EXITED',
-             'Waiting for child threads to terminate...'])
+    expected = [
+        "Bus STOPPING",
+        "Bus STOPPED",
+        "Bus EXITING",
+        "Bus EXITED",
+        "Waiting for child threads to terminate...",
+    ]
+    # If the last message mentions an indeterminable thread name then ignore it
+    assert expected in (log_tracker.log_entries, log_tracker.log_entries[:-1])
```

Comment:
I'm not sure this is reasonable. If there's a bug, it should be fixed. Making the test runner shut up just buries the bug and that is bad.

Reply:
I do not think that it is a bug.  This is about macOS behaving differently than other platforms.  The final message is not returned on macOS but is on other platforms.

Reply:
Interesting... I'm concerned because the related code is probably about a decade old and so if something changed in the env (like newer Pythons or OSs), the code either needs to be updated or it should be made clear in tests to expect one thing in some envs and another thing in different envs. It'd also need some code comment explaining the motivation to prevent regressions.

Reply:
> # If the last message mentions an indeterminable thread name then ignore it

Reply:
Oh, I've probably misread the code and was thinking that you're cutting off https://github.com/cherrypy/cherrypy/blob/4912fca/cherrypy/process/wspbus.py#L352 for some reason.
How about
```suggestion
    expected_len = len(expected)
    assert expected == log_tracker.log_entries[:expected_len]
    assert len(log_tracker.log_entries[expected_len:]) <= 1, (
        'No more than one extra log line with the thread name expected'
    )
```